parquet-hadoop/src/main/java/parquet/hadoop/MemoryManager.java

```diff
@@ -100,18 +112,25 @@ synchronized void removeWriter(InternalParquetRecordWriter writer) {
    */
   private void updateAllocation() {
     long totalAllocations = 0;
-    double scale;
     for (Long allocation : writerList.values()) {
       totalAllocations += allocation;
     }
     if (totalAllocations <= totalMemoryPool) {
-      scale = 1.0;
+      stats.scale = 1.0;
     } else {
-      scale = (double) totalMemoryPool / totalAllocations;
+      stats.scale = (double) totalMemoryPool / totalAllocations;
       LOG.warn(String.format(
           "Total allocation exceeds %.2f%% (%,d bytes) of heap memory\n" +
           "Scaling row group sizes to %.2f%% for %d writers",
-          100*memoryPoolRatio, totalMemoryPool, 100*scale, writerList.size()));
+          100*memoryPoolRatio, totalMemoryPool, 100*stats.scale, writerList.size()));
+      for (Callable callBack : callBacks.values()) {
+        // we do not really want to start a new thread here.
+        try {
+          callBack.call();
+        } catch (Exception e) {
```

Comment:
catch `RuntimeException` not `Exception`

`Exception` includes `InterruptedException` and other things you shouldn't catch by accident.